src/core/meshcop/secure_transport.hpp

```diff
@@ -280,8 +297,108 @@ class Dtls : public InstanceLocator
+#endif // defined(MBEDTLS_SSL_KEEP_PEER_CERTIFICATE)
+
+    /**
+     * Returns an attribute value for the OID 1.3.6.1.4.1.44970.x from the v3 extensions of
+     * the own x509 certificate, where the last digit x is set to aThreadOidDescriptor.
+     * The attribute length is set if the attribute was successfully read or zero if unsuccessful.
+     * Requires a connection to be active.
+     *
+     * @param[in]      aThreadOidDescriptor  The last digit of the Thread attribute OID.
+     * @param[out]     aAttributeBuffer      A pointer to the attribute buffer.
+     * @param[in,out]  aAttributeLength      On input, the size the max size of @p aAttributeBuffer.
+     *                                       On output, the length of the attribute written to the buffer.
+     *
+     * @retval kErrorNone             Successfully read attribute.
+     * @retval kErrorInvalidArgs      Invalid attribute length.
+     * @retval kErrorNotFound         The requested attribute was not found.
+     * @retval kErrorNoBufs           Insufficient memory for storing the attribute value.
+     * @retval kErrorInvalidState     Not connected yet.
+     * @retval kErrorNotImplemented   The value of aThreadOidDescriptor is >127.
+     * @retval kErrorParse            The certificate extensions could not be parsed.
+     *
+     */
+    Error GetThreadAttributeFromOwnCertificate(int      aThreadOidDescriptor,
+                                               uint8_t *aAttributeBuffer,
+                                               size_t  *aAttributeLength);
+
+    /**
+     * Returns an attribute value for the OID 1.3.6.1.4.1.44970.x from the v3 extensions of
+     * the CA x509 certificate chain, where the last digit x is set to aThreadOidDescriptor.
+     * The attribute length is set if the attribute was successfully read or zero if unsuccessful.
+     * Requires a connection to be active.
+     *
+     * @param[in]      aThreadOidDescriptor  The last digit of the Thread attribute OID.
+     * @param[out]     aAttributeBuffer      A pointer to the attribute buffer.
+     * @param[in,out]  aAttributeLength      On input, the size the max size of @p aAttributeBuffer.
+     *                                           On output, the length of the attribute written to the buffer.
+     *
+     * @retval kErrorNone             Successfully read attribute.
+     * @retval kErrorInvalidArgs      Invalid attribute length.
+     * @retval kErrorNotFound         The requested attribute was not found.
+     * @retval kErrorNoBufs           Insufficient memory for storing the attribute value.
+     * @retval kErrorInvalidState     Not connected yet.
+     * @retval kErrorNotImplemented   The value of aThreadOidDescriptor is >127.
+     * @retval kErrorParse            The certificate extensions could not be parsed.
+     *
+     */
+    Error GetThreadAttributeFromCaCertificateChain(int      aThreadOidDescriptor,
```

Comment:
We don't have any use cases for reading Thread-specific OIDs from the CA cert. So I believe this function could be removed. (To avoid confusion when people look into the header files.)

Reply:
fixed.